### PR TITLE
TODO List: Feature list structure

### PR DESCRIPTION
**Pull request by gtekle**
## In this PR, the following are done: 
    1. Set index.html as template page in webpack.
    2. Added webpack.config.js config file
    3. Added tasks object array to index.js 
    4. Added display each task on load event 
    5. Refactored task List Item creating code into a separate function
    6. Added styling and refactor onLoad event handler code into functions
    7. Sorted objects in tasks array based on index property using custom compactor function
    8. Updated readme file
    9. Setup linters and GitHub actions
    10. Fixed linter errors
    11. Setup local dev server as per the Feedback
    12. Updated the readme file's getting started section to guide how to run it.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,47 @@
...
+- [Live Demo Link](https://gtekle.github.io/todo-list/)
+
+## Getting Started
+
+- Install Node.js
+- Install text editor of your choice (eg. VS Code)
+- Clone the repository from [here](https://github.com/gtekle/todo-list.git)
+- Change directory to 'cd todo-list'
+- On the root directory, run 'npm install' on the terminal to install packages.
+- Open the index.html under '/dist' folder file in your browser (using live server) to check your updates and navigate to other pages.
```
Comment (SajjadAhmad14) [inline README.md]: Please add instructions in README to run the project with `webpack-dev-server`.

File: README.md [hunk 2]
```diff
@@ -1 +1,47 @@
...
+
+- HTML, CSS and JavaScript (ES6)
+
+## Live Demo
+
+- [Live Demo Link](https://gtekle.github.io/todo-list/)
+
+## Getting Started
+
+- Install Node.js
```
Comment (SajjadAhmad14) [inline README.md]: Please add a summary to your Pull Request. It can be about project features and how you implemented them.

File: package.json [hunk 3]
```diff
@@ -18,7 +19,19 @@
...
+    "eslint-config-airbnb-base": "^14.2.1",
+    "eslint-plugin-import": "^2.25.4",
+    "hint": "^6.1.9",
+    "html-webpack-plugin": "^5.5.0",
+    "style-loader": "^3.3.1",
+    "stylelint": "^13.13.1",
+    "stylelint-config-standard": "^21.0.0",
+    "stylelint-csstree-validator": "^1.9.0",
+    "stylelint-scss": "^3.21.0",
     "webpack": "^5.65.0",
```
Comment (SajjadAhmad14) [inline package.json:35]: Please add `webpack-dev-server` to the `package.json` file and make sure that it is set up and running with no errors.

File: src/index.js [hunk 4]
```diff
@@ -0,0 +1,83 @@
+import './style.css';
+
+const tasks = [
```
Comment (SajjadAhmad14) [inline src/index.js:3]: Please make sure that the tasks appear on the screen in order of the index values for each task. The user is unable to see a list of tasks on the screen at the moment.

**Review by SajjadAhmad14 — CHANGES_REQUESTED** [review]
## Required Changes ♻️
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hi Tekle, 
 ### You have been doing a tremendous job so far :ok_hand:

## To Highlight 🎉
PR has a title and summary. :heavy_check_mark:
Git flow is followed. :heavy_check_mark:
## To Improve 🙏

- [ ] Please check inline comments under the review

## Optional suggestions
_Comments with the [OPTIONAL] prefix are not a must to implement however, It's strongly recommended to work on them as they can make your code better._

## Happy coding!👏👏👏

#### Please leave any questions or comments in the PR if you have any doubts or unclarity!

**_Note: Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by gtekle** [comment]
Hi @SajjadAhmad14,

Thank you very much for the detailed feedback. You have steered me into the right direction by pointing out a big step I missed on not setting up the local dev server.
I have now corrected it according to the feedback.

Thank you gain! :+1:

**Review by Cathella — APPROVED** [review]
## 🟢 Status: Approved 🎉

### Hi @gtekle 👋🏻, 

You did a tremendous job implementing the requested changes 💪🏻.
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by gtekle** [comment]
Hi @Cathella,
Thank you very much!

**Pull request merged**